controller/lib/hal/hal_stm32_regs.h

```diff
@@ -568,15 +601,37 @@ struct DMA_Regs {
       REG rsvd : 4;
     };
     REG r;
-  } chanSel; // channel selection register
+  } chanSel; // Channel Selection Register [RM] 11.6.7 (pg 317)
 };
 inline DMA_Regs *const DMA1_BASE = reinterpret_cast<DMA_Regs *>(0x40020000);
 inline DMA_Regs *const DMA2_BASE = reinterpret_cast<DMA_Regs *>(0x40020400);
 
-// Select the source for a DMA channel
-// @param dma Address of DMA registers
-// @param chan DMA channel to modify.  Channels are numbered from 0
-// @param selection Selects which peripherial request to map to the channel
+/* Select the source for a DMA channel:
+  @param dma        Address of DMA registers
+  @param chan       DMA channel to modify.  Channels are numbered from 0
+  @param selection  Selects which peripherial request to map to the channel
+
+  DMA request mapping [RM] 11.3.2 (pg 296)
+
+  Channel Selection Register [RM] 11.6.7 (pg 317)
+    Bits 27:24 C7S[3:0]: DMA channel 7 selection
+    Bits 23:20 C6S[3:0]: DMA channel 6 selection
+    Bits 19:16 C5S[3:0]: DMA channel 5 selection
+    Bits 15:12 C4S[3:0]: DMA channel 4 selection
+    Bits 11:8 C3S[3:0]:  DMA channel 3 selection
+    Bits 7:4 C2S[3:0]:   DMA channel 2 selection
+    Bits 3:0 C1S[3:0]:   DMA channel 1 selection
+
+  selection &= 0x0F                 Converts selection to binary and truncates
+                                      digits after 4th bit
+  val = dma->chanSel.r              Holds DMA Channel Selection Register
+  2^(4 * chan)                        MSB position
+  val &= ~(0xF * 2^(4 * chan))      Clears bits storing channel selection on
+  register val |= selection * 2^(4 * chan)   Writes selection to channel
+  selection on register dma->chanSel.r = val              Write changes to
+  channel selection register
+*/
+
```

Comment:
Like elsewhere, I'd like to see if we can tighten this up significantly, maybe into just one line.  Maybe:

```
int x = 4 * static_cast<int>(chan);  // 4 bits per channel ([RM] 11.3.2)
```